### PR TITLE
[Optimization] Avoid unnecessary penalty computation

### PR DESCRIPTION
**Pull request by Sunny-bot1**
## Motivation

重复惩罚计算后处理耗时占比高，但很多时候我们没有传入penalty score，无需进行这些计算

image

## Modifications

- pre-commit
- 判断score的值为默认值时，跳过计算
```
  if (alpha == 1.f && beta == 0.f && gamma == 0.f) {
    return;
  }
```

image

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: custom_ops/gpu_ops/token_penalty_multi_scores.cu [hunk 1]
```diff
@@ -61,36 +61,46 @@ __global__ void update_repeat_times(const int64_t *pre_ids,
...
-    int64_t bi = blockIdx.x;
-    if (cur_len[bi] < 0) {
-        return;
+  int64_t bi = blockIdx.x;
+  float alpha = penalty_scores[bi];
+  float beta = frequency_score[bi];
+  float gamma = presence_score[bi];
+  if (alpha == 1.f && beta == 0.f && gamma == 0.f) {
+    return;
+  }
```
Comment (Sunny-bot1) [inline custom_ops/gpu_ops/token_penalty_multi_scores.cu:77]: 修改1

File: custom_ops/gpu_ops/token_penalty_multi_scores.cu [hunk 2]
```diff
@@ -103,25 +113,29 @@ __global__ void update_value_by_repeat_times(const int *repeat_times,
...
+  T *logits_now = logits + bi * vocab_size;
+  const int *repeat_times_now = repeat_times + bi * vocab_size;
+  const int *is_repeated_now = is_repeated + bi * vocab_size;
+  float alpha = static_cast<float>(penalty_scores[bi]);
+  float beta = static_cast<float>(frequency_score[bi]);
+  float gamma = static_cast<float>(presence_score[bi]);
+  float temperature = temperatures[bi];
+  if (alpha == 1.f && beta == 0.f && gamma == 0.f && temperature == 1.f) {
+    return;
+  }
```
Comment (Sunny-bot1) [inline custom_ops/gpu_ops/token_penalty_multi_scores.cu:127]: 修改2

**Review by zhoutianzi666 — APPROVED** [review]
